homeassistant/components/sensor/pollen.py

```diff
@@ -0,0 +1,349 @@
+        {'key': 'Yesterday'},
+        'mdi:flower'
+    ),
+    'disease_average_forecasted': (
+        'Cold & Flu: Forecasted Average',
+        'AllergyAverageSensor',
+        'disease_average_data',
+        {'data_attr': 'extended_data'},
+        'mdi:snowflake'
+    )
+}
+
+RATING_MAPPING = [{
+    'label': 'Low',
+    'minimum': 0.0,
+    'maximum': 2.4
+}, {
+    'label': 'Low/Medium',
+    'minimum': 2.5,
+    'maximum': 4.8
+}, {
+    'label': 'Medium',
+    'minimum': 4.9,
+    'maximum': 7.2
+}, {
+    'label': 'Medium/High',
+    'minimum': 7.3,
+    'maximum': 9.6
+}, {
+    'label': 'High',
+    'minimum': 9.7,
+    'maximum': 12
+}]
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ZIP_CODE): cv.positive_int,
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(CONDITIONS)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Configure the platform and add the sensors."""
+    from pypollencom import Client
+
+    _LOGGER.debug('Configuration data: %s', config)
+
+    client = Client(config[CONF_ZIP_CODE])
+    datas = {
+        'allergy_average_data': AllergyAveragesData(client),
+        'allergy_index_data': AllergyIndexData(client),
+        'disease_average_data': DiseaseData(client)
+    }
+
+    for data in datas.values():
+        data.update()
+
+    sensors = []
+    for condition in config.get(CONF_MONITORED_CONDITIONS, []):
+        name, sensor_class, data_key, params, icon = CONDITIONS[condition]
+        sensors.append(globals()[sensor_class](
+            datas[data_key],
+            params,
+            name,
+            icon
+        ))
+
+    add_devices(sensors, True)
+
+
+def average_of_list(list_of_nums, decimal_places=1):
+    """Return the average of a list of ints."""
+    return round(sum(list_of_nums, 0.0)/len(list_of_nums), decimal_places)
+
+
+def calculate_trend(list_of_nums):
+    """Return the average of a list of ints."""
+    ratings = list(
+        map(
+            (lambda n: [
+                r['label'] for r in RATING_MAPPING
+                if r['minimum'] <= n <= r['maximum']
+            ][0]),
+            list_of_nums
+        )
+    )
+    return max(set(ratings), key=ratings.count)
+
+
+def merge_two_dicts(dict1, dict2):
+    """Merge two dicts into a new dict as a shallow copy."""
+    final = dict1.copy()
+    final.update(dict2)
+    return final
+
+
+class BaseSensor(Entity):
+    """Define a base class for all of our sensors."""
+
+    def __init__(self, data, data_params, name, icon):
+        """Initialize the sensor."""
+        self._attrs = {}
+        self._icon = icon
+        self._name = name
+        self._data_params = data_params
+        self._state = None
+        self._unit = None
+        self.data = data
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return merge_two_dicts(
```

Comment:
Why do you need this? Can't you just update the attributes?

Reply:
I'm merging common attributes of the base class (e.g. `ATTR_ATTRIBUTION`) why (pseudo-)dynamic attributes of the subclasses. I've done this before; not good?

Reply:
Why do you need to copy before updating? You can just do:
```py
self._attr.update({...})
return self._attr
```

Reply:
Oh wow, silly on me. Thanks!